yjit/src/codegen.rs

```diff
@@ -246,6 +251,28 @@ pub enum JCCKinds {
     JCC_JNA,
 }
 
+/// Counter used by get_counted_exit to count the number of exits with --yjit-stats.
+#[derive(Clone)]
+struct ExitCounter {
+    ptr: *const u8,
+    name: String,
+}
+
+/// Macro to get an ExitCounter for helpers with side exits.
+macro_rules! get_counter {
+    ($counter_name:ident) => {
+        if (get_option!(gen_stats)) {
+            let counter = ExitCounter {
+                ptr: ptr_to_counter!($counter_name) as *const u8,
+                name: stringify!($counter_name).to_string(),
+            };
+            Some(counter)
+        } else {
+            None
+        }
+    };
+}
```

Comment:
It feels hacky to introduce this auxiliary struct and macro when we already have a way to refer to counters by name string. Maybe we could just use `Option(String)` instead?

Eventually we should probably generate a Rust enum for the counters.

Reply:
> we already have a way to refer to counters by name string

Do we? `ptr_to_counter` takes identifier and not a String, so I'm not sure if we have a macro for it.

Reply:
Oh, I found `get_counter_ptr`. I think I can use it.

Reply:
kk, will try to fix that later.

Reply:
Using `ptr_to_counter!` has advantage that a typo is going to be compile-time warning rather than a runtime panic and it allows you to rename a symbol easily, but I'm okay with using a String if people prefer it.

Reply:
> Ultimately I think this points to the stat counters should be an enum

Agreed :+1: Let's revisit this later.